In [1]:
import cv2 as cv

take path to an img and return a matrix of pixels

In [3]:
img = cv.imread('./Resources/Photos/cat.jpg')
img.shape

(427, 640, 3)

to show img, pass the name of window and matrix want to show

In [4]:
cv.imshow('Cat', img)

since img close instantly

In [ ]:
cv.waitKey(0)

opencv has no built-in fn to handle img greater than screen

reading video

create a capture var, pass the path of video file, or int like 0 ,1, 2, ... if using webcam or camera connection to computer.

webcam referred to as 0,

In [2]:
capture = cv.VideoCapture('./Resources/Videos/dog.mp4')

use a loop, read frame by frame

In [3]:
while True:
    isTrue, frame = capture.read()
    cv.imshow('Video', frame)

    if cv.waitKey(20) and 0xFF == ord('d'):
        break

capture.release()
cv.destroyAllWindows()

error: OpenCV(4.12.0) D:\a\opencv-python\opencv-python\opencv\modules\highgui\src\window.cpp:973: error: (-215:Assertion failed) size.width>0 && size.height>0 in function 'cv::imshow'


-215 insertion error， in most cases, means opencv can't find file in the location, b/c ran out of frame